pkg/osbuild/oscap_autotailor_stage.go

```diff
@@ -11,29 +11,57 @@ type OscapAutotailorStageOptions struct {
 	Config   OscapAutotailorConfig `json:"config"`
 }
 
-type OscapAutotailorConfig struct {
-	TailoredProfileID string   `json:"new_profile"`
-	Datastream        string   `json:"datastream"`
-	ProfileID         string   `json:"profile_id"`
-	Selected          []string `json:"selected,omitempty"`
-	Unselected        []string `json:"unselected,omitempty"`
+type OscapAutotailorConfig interface {
+	validate() error
+	isAutotailorConfig()
 }
 
-func (OscapAutotailorStageOptions) isStageOptions() {}
+type AutotailorKeyValueConfig struct {
+	NewProfile string   `json:"new_profile,omitempty"`
+	Datastream string   `json:"datastream"`
+	ProfileID  string   `json:"profile_id"`
+	Selected   []string `json:"selected,omitempty"`
+	Unselected []string `json:"unselected,omitempty"`
+}
+
+func (c AutotailorKeyValueConfig) isAutotailorConfig() {}
 
-func (c OscapAutotailorConfig) validate() error {
+func (c AutotailorKeyValueConfig) validate() error {
 	if c.Datastream == "" {
 		return fmt.Errorf("'datastream' must be specified")
 	}
+	if c.NewProfile == "" {
+		return fmt.Errorf("'new_profile' must be specified")
+	}
 	if c.ProfileID == "" {
-		return fmt.Errorf("'profile_id' must be specified")
+		return fmt.Errorf("'profile_id' tailoring_file' must be specified")
+	}
+	return nil
+}
+
+type AutotailorJSONConfig struct {
+	TailoredProfileID string `json:"tailored_profile_id,omitempty"`
+	Datastream        string `json:"datastream"`
+	TailoringFile     string `json:"tailoring_file,omitempty"`
```

Comment:
All of these are required by the stage schema, so shouldn't be really `omitempty`.